tests/plugins/ip_public_test.py

```diff
@@ -0,0 +1,54 @@
+import pytest
+
+from detect_secrets.plugins.ip_public import IPPublicDetector
+
+
+class TestIPPublicDetector:
+
+    class TestIPv4:
+        """
+        Testing strategy
+
+        Cover the cartesian product of these partitions:
+
+          1. Partition on ip address format:
+              a. Valid ipv4 address
+
+          2. Partition on ip address type:
+              a. Public
+              b. Non-public
+
+        And cover this case:
+          1. Partition on ip address format:
+              a. Invalid ipv4 address
+        """
+
+        @pytest.mark.parametrize(
+            'payload, should_flag',
+            [
+                # Valid IPv4 addresses, Public
+                ('133.133.133.133', True),
+                ('This line has an IP address 133.133.133.133@something else', True),
+                ('133.133.133.133:8080', True),
+                ('This line has an IP address: 133.133.133.133:8080@something else', True),
+                ('1.1.1.1', True),
+                # Valid IPv4 addresses, Non-public
+                ('127.0.0.1', False),
+                ('10.0.0.1', False),
+                ('172.16.0.1', False),
+                ('192.168.0.1', False),
+                # Invalid IPv4 addresses
+                ('256.256.256.256', False),
+                ('1.2.3', False),
+                ('1.2.3.4.5.6', True),
+                ('1.2.3.4.5.6.7.8', '2'),
+                ('1.2.3.04', True),
```

Comment:
This one is also an invalid IPv4 address, but differently from the two cases above it follows the generic format correctly. This case (I think) should return true, but I'd like to ask for @jpdakran for his opinion on this

Reply:
@perryzjc after thinking about this a bit more, I think an invalid IP should always return `False`, therefore this case should return `False` too (this in an attempt to avoid false positives)

Reply:
> @perryzjc after thinking about this a bit more, I think an invalid IP should always return `False`, therefore this case should return `False` too (this in an attempt to avoid false positives)

@lorenzodb1 I feel it makes sense. Just adjusted the test case and the implementation.